plugins/woocommerce-admin/client/customize-store/assembler-hub/sidebar/navigation-extra-screen/sidebar-navigation-extra-screen.tsx

```diff
@@ -0,0 +1,56 @@
+/**
+ * External dependencies
+ */
+import { useQuery } from '@woocommerce/navigation';
+import { useMemo } from '@wordpress/element';
+
+/**
+ * Internal dependencies
+ */
+import { SidebarPatternScreen } from '../pattern-screen/sidebar-pattern-screen';
+
+import './style.scss';
+import { PATTERN_CATEGORIES } from '../pattern-screen/categories';
+
+const getPatternCategoryByPath = ( path: string ) => {
```

Comment:
Why aren't we basing this implementation on Gutenberg's [Inserter Sidebar](https://href.li/?https://github.com/WordPress/gutenberg/blob/550cb6b67b544c72d124e8d6741e0d5577da9256/packages/editor/src/components/inserter-sidebar/index.js#L90) and [InserterLibrary](https://github.com/WordPress/gutenberg/blob/15b8049af46b1849e96bc52c8d99386231bcc558/packages/block-editor/src/components/inserter/library.js#L15) ?

Reply:
Yes! The `InsertSidebar` isn't a public component, but wraps the `InsertLibrary`. This latter, instead, doesn't allow us to hide the header and render only the pattern categories. Futhermore, even if we are able to do it, that component renders ALL the pattern categories that are registered; instead, we want to render only some of them.

Reply:
WDYT about proposing upstream changes to support the usage of `InserterLibrary` in this particular case? There's a concern from the leadership about duplicating work and creating a "new editor" so I would recommend at least exploring this approach before proceeding with creating our dedicated component instead. Reference: pdnLyh-6dS-p2#comment-4563

Reply:
Good observation! Thanks for pointing this out! 

> WDYT about proposing upstream changes to support the usage of InserterLibrary in this particular case?


IMHO, a tailored inserter with only specific categories isn't in the goal of Gutenberg: even if it is in "simple mode," by definition, Site Editor has to be extendible by 3rd party plugins, so I think that it is safe to assume that all the categories should always be visible.



Also, I don't see a lot of duplicate work here because this component is just a wrapper around the `BlockPatternList` + custom handlers.